### PR TITLE
[WIP] Opgops 2005

### PR DESCRIPTION
**Pull request by brettminnie**

### CODE REVIEW

File: roles/admins/tasks/users.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        home: "/home/{{ user }}"
+        shell: /bin/bash
+        groups: wheel, adm, root
+
+    - name: Create required user directories
+      file:
+        path: "{{ item }}"
+        state: directory
+        owner: "{{ user.name }}"
+        mode: 0600
```
Comment (tim-mcarthur-moj) [inline roles/admins/tasks/users.yml]: 700 is the correct mode for these dirs, with 600 for files

File: roles/common/tasks/directories.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+
+- name: Create required system directories
+  file:
+    path: "{{ item }}"
+    state: directory
+  with_items:
+    - /data
+    - /srv
+    - /var/log
```
Comment (tim-mcarthur-moj) [inline roles/common/tasks/directories.yml]: surely we don't need this?

File: roles/cron/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+---
+
+- name: Set cron_paths fact
+  set_fact:
+    cron_paths: "/usr/local/bin:/usr/bin:/bin"
```
Comment (tim-mcarthur-moj) [inline roles/cron/tasks/main.yml]: are `sbin` directories not also needed in path

File: roles/docker-service/README.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
+docker-service
+--------------
+
+This module is an extension of the docker compose formula allowing the formula to deliver an ecs service or a docker compose file.
```
Comment (tim-mcarthur-moj) [inline roles/docker-service/README.md]: needs updating to reflect how it is used with ansible

**Review by tim-mcarthur-moj — CHANGES_REQUESTED** [review]

File: roles/cron/tasks/main.yml (hunk 3 above)
Comment (brettminnie) [reply roles/cron/tasks/main.yml]: Good call

File: roles/docker-service/README.md (hunk 4 above)
Comment (brettminnie) [reply roles/docker-service/README.md]: This entire role needs to be updated to play nicely with ansilble, was committed so I don't lose it while I migrate all the other things

File: roles/common/tasks/directories.yml (hunk 2 above)
Comment (brettminnie) [reply roles/common/tasks/directories.yml]: Are`srv/` and `data/` auto cucumbered?

I agree with `/var/log`

File: filter_plugins/opg_custom.py [hunk 5]
```diff
@@ -113,6 +113,14 @@ def find_rds_instance(rds_list, db_name):
     return db_info
 
 
+def update(*dicts):
+    res_dict = {}
```
Comment (carinadigital) [inline filter_plugins/opg_custom.py]: Would be nice to have a better name and sentence describing what this does e.g.
``` combine_dicts()```
``` Takes a list of dictionaries and returns a combined dictionary. keys are overwritten is present in more that one dictionary```

File: roles/admins/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+---
+
+- name: Manage admin user
+  include: "users.yml"
+  with_items: "{{ admins | d([]) }}"
```
Comment (carinadigital) [inline roles/admins/tasks/main.yml]: What do you think about using longform  `default()` everywhere?

File: roles/bootstrap/files/bash-prompt.sh [hunk 7]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+if [[ "${OPG_ENVIRONMENT}" =~ "production" ]]
+then
+    PS1="\[\033[01;31m\](${OPG_ENVIRONMENT}-${OPG_STACKNAME})\[\e[m\]\[\033[01;32m\]$PS1\[\033[00m\] "
```
Comment (carinadigital) [inline roles/bootstrap/files/bash-prompt.sh]: Can you comment what type of prompt this produces?

File: roles/bootstrap/tasks/hostname.yml [hunk 8]
```diff
@@ -0,0 +1,43 @@
+---
+
+- name: define our metadata url
+  set_fact:
+    ec2_metadata_url: 'http://169.254.169.254/latest/meta-data'
```
Comment (carinadigital) [inline roles/bootstrap/tasks/hostname.yml]: Isn't this fact already in `common` role above?

File: roles/bootstrap/tasks/packages.yml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+  apt:
+    update_cache: yes
+  become: true
+
+- name: Install base packages
+  apt:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - joe
```
Comment (carinadigital) [inline roles/bootstrap/tasks/packages.yml]: why `joe`?

File: roles/collectd/README.md [hunk 10]
```diff
@@ -0,0 +1,58 @@
+###collectd
```
Comment (carinadigital) [inline roles/collectd/README.md]: Could we not have leveraged something like https://github.com/azavea/ansible-collectd ?

**Review by carinadigital — CHANGES_REQUESTED** [review]
I think it could be better to split out new roles in separate PRs, that can be reviewed an merged earlier

File: roles/admins/tasks/main.yml (hunk 6 above)
Comment (Azulinho) [reply roles/admins/tasks/main.yml]: +1

File: roles/bootstrap/tasks/packages.yml (hunk 9 above)
Comment (brettminnie) [reply roles/bootstrap/tasks/packages.yml]: was a default from old systems, I did ask if it was required, no-one answered me so I left it in

File: roles/bootstrap/tasks/hostname.yml (hunk 8 above)
Comment (brettminnie) [reply roles/bootstrap/tasks/hostname.yml]: Good spot, I must have missed this

File: roles/collectd/README.md (hunk 10 above)
Comment (brettminnie) [reply roles/collectd/README.md]: Based on our opg-collectd formula, I'll check the differences

File: roles/bootstrap/files/bash-prompt.sh (hunk 7 above)
Comment (Azulinho) [reply roles/bootstrap/files/bash-prompt.sh]: I was about to say this is baked in the AMI, but remembered 1701 is not merged yet. so will keep my mouth shut

Comment (brettminnie) [reply roles/bootstrap/files/bash-prompt.sh]: It is and it isn't, the version baked into the ami breaks on display somehow, so I tweaked it, we need to consolidate the two @Azulinho

File: roles/bootstrap/tasks/hostname.yml (hunk 8 above)
Comment (brettminnie) [reply roles/bootstrap/tasks/hostname.yml]: Upon testing this is not picking it up from the previous scope, I've left it redeclared for now

**Comment by brettminnie** [comment]
Will be closed when sub prs are merged in

**Comment by brettminnie** [comment]
Closed via #83, #84, #85,#86, #87, #88, #89, #90, #91, #92, #93, #94

**Pull request closed without merging**